SwiftWisdom/Core/StandardLibrary/Numbers/UnsignedInteger+Extensions.swift

```diff
@@ -8,38 +8,38 @@
 
 import Foundation
 
+// REVIEW: can we test this? how? 
 extension UnsignedInteger {
     public static func ip_random() -> Self {
         let intMax = Int(ip_maxValue.toIntMax())
-        let rand = randomInRange(0...intMax)
+        let rand = random(inRange: 0...intMax)
         return self.init(ip_safely: rand)
     }
 }
 
-public func randomInRange(_ range: ClosedRange<Int>) -> Int {
-    // TODO: Not good API design - since closed range does not assume positive only. We should either
-    // a) rename method to indicate positive only
-    // b) remove lower bound restriction
-    assert(range.lowerBound >= 0)
+// REVIEW: should this be moved into Int+Extension because it uses Int ?
+// if yes dont forget to move its respective test
+public func random(inRange range: CountableClosedRange<Int>) -> Int {
     let diff = range.upperBound - range.lowerBound
     let randomOffset = Int(arc4random_uniform(UInt32(diff + 1)))
     let random = range.lowerBound + randomOffset
     return random
 }
 
+// TODO: write tests for this extension
 extension UnsignedInteger {
+    // FIXME: drop NS and just use Data, requires NSData+Extension to be updated first
```

Comment:
Actually I'd like to keep the NSData initializer and just add another one that takes Data.